app.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python
+import os
+import random
+"""
+    This module implements Amity as a class based interface for the entire
+    room allocation system
+"""
+
+from models.person import Fellow, Staff
+from models.room import Office, LivingSpace
+
+class Amity(object):
+
+    def __init__(self):
+        self.room_directory = []
+        self.waiting_list = []
+        self.living_waiting_list = []
+
+    def add_person(self, fname, sname, role="none", wants_living = "N"):
+        """ method to add people to system utilising models"""
+        if isinstance(fname, str) and isinstance(sname, str):
+            if role.lower() == "fellow":
+                person = Fellow(fname, sname, role, wants_living)
+            elif role.lower() == "staff":
+                if wants_living == "Y":
+                    return "Staff aren't eligible for accomodation"
+                else:
+                    person = Staff(fname, sname)
+            else:
+                return "the role is invalid"
+            self.waiting_list.append(person)
+        elif not isinstance(fname, str) or not isinstance(sname, str):
+            return "the name is invalid"
+
+    def add_room(self, rm_type, *given_names):
+        """ method to add rooms to system utilising models specified """
+        for rm_name in given_names:
+            if isinstance(rm_type, str) and isinstance(rm_name, str):
+                rm_variable =  rm_name.lower()
+                if rm_variable not in [room.name for room in self.room_directory]:
+                    if rm_type == "l":
+                        room = LivingSpace(rm_variable)
+                    elif rm_type == "o" or "office":
+                        room = Office(rm_variable)
+                    else:
+                        print ("the option given to create room is invalid")
+                    new_room = [room]
+                    self.room_directory.extend(new_room)
+                else:
+                    return "the room already exists"
+            elif not isinstance(rm_type, str):
+                return "the room option is invalid"
+            elif not isinstance(rm_name, str):
+                return "the room name is invalid"
+
+    def get_available_space(self, room_type):
+        """ filter the room directory to return available rooms for occupancy """
+        if room_type == "o":
+            office_w_space = [room for room in self.room_directory
+                if (len(room.occupants)< room.capacity) and isinstance(room, Office)]
+            return office_w_space
```

Comment:
The indention of `return office_w_space` is quite suspicious, are you sure it is working and you are getting what you intend?